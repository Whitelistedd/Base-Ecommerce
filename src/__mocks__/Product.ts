export const TEST_PRODUCT1 = {
	categories: ['hoodies'],
	color: ['black'],
	createdAt: '2022-06-16T11:18:10.448Z',
	desc: '',
	gender: ['men'],
	img: [
		'https://i.ibb.co/93ksW04/Untitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp',
	],
	inStock: 8,
	price: 3000,
	size: ['XL', 'L', 'M'],
	title: "MEN'S QUARTER ZIP HOODIE",
	updatedAt: '2022-06-16T11:18:10.448Z',
	__v: 0,
	_id: '62ab1172bb9ddb4cfa932c50',
}
export const TEST_PRODUCT2 = {
	categories: ['hoodies'],
	color: ['black'],
	createdAt: '2022-06-16T11:18:10.448Z',
	desc: '',
	gender: ['men'],
	img: [
		'https://i.ibb.co/93ksW04/Untitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp',
	],
	inStock: 8,
	price: 3000,
	size: ['XL', 'L', 'M'],
	title: "MEN'S QUARTER ZIP HOODIE",
	updatedAt: '2022-06-16T11:18:10.448Z',
	__v: 0,
	_id: '62ab1172bb9ddb4cfa932c51',
}
export const TEST_PRODUCT3 = {
	categories: ['hoodies'],
	color: ['black'],
	createdAt: '2022-06-16T11:18:10.448Z',
	desc: '',
	gender: ['men'],
	img: [
		'https://i.ibb.co/93ksW04/Untitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp',
	],
	inStock: 8,
	price: 3000,
	size: ['XL', 'L', 'M'],
	title: "MEN'S QUARTER ZIP HOODIE",
	updatedAt: '2022-06-16T11:18:10.448Z',
	__v: 0,
	_id: '62ab1172bb9ddb4cfa932c52',
}
