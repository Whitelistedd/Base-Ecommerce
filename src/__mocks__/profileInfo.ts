const profileInfo = {
  firstName: 'test',
  lastName: 'test',
  address: {
    address: 'testAddress',
    city: 'testCity',
    apartment: 'testApartment',
    zipCode: 'testZipCode',
    country: '',
  },
}

export default profileInfo
