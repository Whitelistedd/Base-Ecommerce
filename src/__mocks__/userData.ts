const userData = {
	name: 'user',
	email: 'user@example.com',
}

export default userData
