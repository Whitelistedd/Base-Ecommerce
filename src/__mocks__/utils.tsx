/* eslint-disable react/display-name */

import * as React from 'react'

import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

import { render } from '@testing-library/react'

const createTestQueryClient = () =>
	new QueryClient({
		defaultOptions: {
			queries: {
				retry: false,
			},
		},
		logger: {
			log: console.log,
			warn: console.warn,
			// eslint-disable-next-line @typescript-eslint/no-empty-function
			error: () => {},
		},
	})

export function renderWithClient(client: QueryClient, ui: React.ReactElement) {
	const { rerender, ...result } = render(
		<QueryClientProvider client={client}>{ui}</QueryClientProvider>
	)
	return {
		...result,
		rerender: (rerenderUi: React.ReactElement) =>
			rerender(
				<QueryClientProvider client={client}>{rerenderUi}</QueryClientProvider>
			),
	}
}

export function createWrapper() {
	const testQueryClient = createTestQueryClient()
	return ({ children }: { children: React.ReactNode }) => (
		<QueryClientProvider client={testQueryClient}>
			{children}
		</QueryClientProvider>
	)
}
