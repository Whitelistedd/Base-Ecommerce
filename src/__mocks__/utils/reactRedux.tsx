import { Provider } from 'react-redux'
import cartSlice from '@/redux/slices/cart'
import { configureStore } from '@reduxjs/toolkit'
import userSlice from '@/redux/slices/user'

export const mockStore = configureStore({
  reducer: {
    user: userSlice,
    cart: cartSlice,
  },
})

export function wrapWithRedux(children: React.ReactNode) {
  return <Provider store={mockStore}>{children}</Provider>
}
