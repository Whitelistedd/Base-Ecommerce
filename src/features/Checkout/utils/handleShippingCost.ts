/* расчет доставки, чтобы показать пользователю сумму */
export const handleShipping = (
  e: React.MouseEvent<HTMLInputElement, MouseEvent>
) => {
  const target = e.target as HTMLInputElement
  if (target.value === 'PochtaRussia') {
    return 500
  } else if (target.value === 'Cdek') {
    return 700
  } else {
    return 0
  }
}
