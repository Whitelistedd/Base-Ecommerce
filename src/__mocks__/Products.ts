import { TEST_PRODUCT1, TEST_PRODUCT2, TEST_PRODUCT3 } from './Product'

const TEST_PRODUCTS = [TEST_PRODUCT1, TEST_PRODUCT2, TEST_PRODUCT3]

export default TEST_PRODUCTS
