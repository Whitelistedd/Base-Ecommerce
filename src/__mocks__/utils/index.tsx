import { Provider } from 'react-redux'
import { QueryClientProvider } from '@tanstack/react-query'
import { UserProvider } from '@auth0/nextjs-auth0'
import { createTestQueryClient } from './reactQuery'
import { mockStore } from './reactRedux'

export function wrapWithAll(children: React.ReactNode) {
  const testQueryClient = createTestQueryClient()
  return (
    <Provider store={mockStore}>
      <QueryClientProvider client={testQueryClient}>
        <UserProvider>{children}</UserProvider>
      </QueryClientProvider>
    </Provider>
  )
}
