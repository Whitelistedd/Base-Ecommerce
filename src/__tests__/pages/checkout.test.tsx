import { AnyAction, configureStore } from '@reduxjs/toolkit'

import CheckoutPage from '@/pages/checkout'
import { Provider } from 'react-redux'
import { UserProvider } from '@auth0/nextjs-auth0'
import cartSlice from '@/redux/slices/cart'
import renderer from 'react-test-renderer'
import userSlice from '@/redux/slices/user'
import { wrapWithClient } from '@/__mocks__/utils'

jest.mock('next/router', () => require('next-router-mock'))

describe('Checkout  page', () => {
  const mockStore = configureStore({
    reducer: {
      user: userSlice,
      cart: cartSlice,
    },
  })
  it('should match snapshot', () => {
    const tree = renderer
      .create(
        wrapWithClient(
          <Provider store={mockStore}>
            <UserProvider>
              <CheckoutPage />
            </UserProvider>
          </Provider>
        )
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
