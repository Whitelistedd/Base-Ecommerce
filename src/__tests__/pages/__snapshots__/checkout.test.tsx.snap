// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout  page should match snapshot 1`] = `
<div
  className="sc-283eddd0-1 gkkIUt"
>
  <div
    className="sc-283eddd0-0 dZUPki"
  >
    <div
      className="sc-92718e0b-2 hhAWhF"
    >
      <div
        className="sc-92718e0b-1 iDnurs"
      >
        <div
          className="sc-92718e0b-0 hbQEh"
        >
          <span
            style={
              {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "opacity": 1,
                  "padding": 0,
                  "paddingTop": "100%",
                  "width": "initial",
                }
              }
            />
            <img
              alt="loading..."
              className="sc-92718e0b-3"
              data-nimg="responsive"
              decoding="async"
              onError={[Function]}
              onLoad={[Function]}
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                alt="loading..."
                className="sc-92718e0b-3"
                data-nimg="responsive"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=%2Fassets%2Fimages%2Floading.webp&w=3840&q=75"
                srcSet="/_next/image?url=%2Fassets%2Fimages%2Floading.webp&w=640&q=75 640w, /_next/image?url=%2Fassets%2Fimages%2Floading.webp&w=750&q=75 750w, /_next/image?url=%2Fassets%2Fimages%2Floading.webp&w=828&q=75 828w, /_next/image?url=%2Fassets%2Fimages%2Floading.webp&w=1080&q=75 1080w, /_next/image?url=%2Fassets%2Fimages%2Floading.webp&w=1200&q=75 1200w, /_next/image?url=%2Fassets%2Fimages%2Floading.webp&w=1920&q=75 1920w, /_next/image?url=%2Fassets%2Fimages%2Floading.webp&w=2048&q=75 2048w, /_next/image?url=%2Fassets%2Fimages%2Floading.webp&w=3840&q=75 3840w"
                style={
                  {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-c244e443-7 ijryVv"
  >
    <div
      className="sc-c244e443-0 hENCiU"
    >
      <div
        className="sc-c244e443-1 cBLjNq"
      >
        <div
          className="sc-c244e443-2 fugwvk"
        >
          <p
            className="sc-c244e443-3 dTgOXL"
          >
            Subtotal
          </p>
          <p
            className="sc-c244e443-4 cMlBYJ"
          >
            ₽
            0
          </p>
        </div>
        <div
          className="sc-c244e443-2 fugwvk"
        >
          <p
            className="sc-c244e443-3 dTgOXL"
          >
            Shipping
          </p>
          <p
            className="sc-c244e443-4 cMlBYJ"
          >
            ₽
            500
          </p>
        </div>
      </div>
      <div
        className="sc-c244e443-2 fugwvk"
      >
        <p
          className="sc-c244e443-5 cDJHPh"
        >
          Total
        </p>
        <p
          className="sc-c244e443-6 kHUEiq"
        >
          ₽
          500
        </p>
      </div>
    </div>
  </div>
</div>
`;
